### PR TITLE
fix: generate types from JSDoc

### PR DESCRIPTION
**Pull request by wkillerud**
Expand the typing/JSDoc internally in the module as well

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -91,15 +121,15 @@ export default class PodiumClient extends EventEmitter {
...
         this.#metrics = new Metrics();
-        this.#metrics.on('error', error => {
+        this.#metrics.on('error', (error) => {
             log.error(
                 'Error emitted by metric stream in @podium/client module',
                 error,
             );
         });
 
         this[Symbol.iterator] = () => ({
```
Comment (wkillerud) [inline lib/client.js:131]: I don't think this `Symbol.iterator` API is possible to type

File: lib/resolver.content.js [hunk 2]
```diff
@@ -227,6 +252,7 @@ export default class PodletClientContentResolver {
             if (outgoing.redirectable && statusCode >= 300) {
                 outgoing.redirect = {
                     statusCode,
+                    // @ts-expect-error TODO: look into what happens if the podlet returns more than one location header
```
Comment (wkillerud) [inline lib/resolver.content.js:255]: TS highlighted this as a potential error, but I didn't want to make a decision there now

File: lib/utils.js [hunk 3]
```diff
@@ -21,41 +19,35 @@ export const isHeaderDefined = (headers, header) => {
...
+export const validateIncoming = (incoming = {}) =>
+    Object.prototype.toString.call(incoming) === '[object PodiumHttpIncoming]';
 
 /**
  * Filter assets array based on scope.
  * If scope property is not present, asset will be included (backwards compatibility)
  * If scope property is set to "all", asset will be included.
  * If scope is set to "content" and asset scope property is set to "fallback", asset will not be included
  * If scope is set to "fallback" and asset scope property is set to "content", asset will not be included
+ * @template {import("@podium/utils").AssetCss | import("@podium/utils").AssetJs} T[]
```
Comment (wkillerud) [inline lib/utils.js:47]: Replaced with generics so the type is kept untouched on the `Response` class itself

**Comment by wkillerud** [comment]
Tested with local linking via `@podium/layout`, and it looks like the types are working as expected.

**Review by trygve-lie — APPROVED** [review]
lgtm :+1:

**Pull request merged**